src/libraries/System.Private.CoreLib/src/System/Resources/RuntimeResourceSet.cs

```diff
@@ -353,14 +355,11 @@ private IDictionaryEnumerator GetEnumeratorHelper()
             return value;
         }
 
-        private static object? ReadValue (ResourceReader reader, int dataPos, bool isString, out ResourceLocator locator)
+        private static object? ReadValue(ResourceReader reader, int dataPos, bool isString, out ResourceLocator locator)
         {
             object? value;
             ResourceTypeCode typeCode;
 
-            // Normally calling LoadString or LoadObject requires
-            // taking a lock.  Note that in this case, we took a
-            // lock before calling this method.
```

Comment:
This comment was not (no longer?) correct. We do typically call these within a lock, but it is a lock on either `_resCache` or `_caseInsensitiveTable`. That lock was previously making things thread-safe so long as the caller was only using `GetString/GetObject` and was not flipping `ResourceManager.IgnoreCase` on and off. 

Now, `LoadString`/`LoadObject` _do_ lock internally so there is no edge case here.